object_store/src/aws/mod.rs

```diff
@@ -1094,12 +1103,36 @@ impl AmazonS3Builder {
     }
 }
 
+#[cfg(feature = "aws_profile")]
+fn profile_region(profile: String) -> Option<String> {
+    use std::{panic, thread};
+    use tokio::runtime::Handle;
+
+    let handle = Handle::current();
+    let provider = profile::ProfileProvider::new(profile, None);
+
+    let result = thread::spawn(move || handle.block_on(provider.get_region()));
+
+    match result.join() {
+        Ok(region) => region,
+        Err(e) => panic::resume_unwind(e),
+    }
```

Comment:
```suggestion
    handle.block_on(provider.get_region())
```

Both are equally "problematic" in that they pin a tokio worker, but this at least avoids spawning an additional thread